### PR TITLE
feat: integrate payload service

### LINKED ISSUE #2219: Phase out `PayloadStore` trait with new types
### Describe the feature

replace usage of https://github.com/paradigmxyz/reth/blob/703f0fdf8904dfb789019deeeddc57813cc8ec90/crates/payload/builder/src/lib.rs#L37-L37 with new service and handler types

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mattsse**
Closes #2219

replaces `PayloadStore` with new builder service and handle

This needs two followups:

* move handling of `getPayload` to engine API handler, doesn't need to be in beaconconsensus engine
* support async return of new payload id

### CODE REVIEW

File: crates/consensus/beacon/src/engine/mod.rs [hunk 1]
```diff
@@ -279,18 +279,20 @@ where
...
-        match self.payload_store.get_execution_payload(payload_id) {
-            Some(payload) => Ok(payload),
-            None => Err(EngineRpcError::UnknownPayload.into()),
-        }
+        // match self.payload_builder.get_execution_payload(payload_id) {
+        //     Some(payload) => Ok(payload),
+        //     None => Err(EngineRpcError::UnknownPayload.into()),
+        // }
+
+        todo!()
```
Comment (Rjected) [inline crates/consensus/beacon/src/engine/mod.rs:295]: Makes sense given we want to move it

**Review by Rjected — APPROVED** [review]
LGTM

File: crates/consensus/beacon/src/engine/mod.rs [hunk 2]
```diff
@@ -279,18 +279,20 @@ where
...
 
         // for now just return the output from the payload store
-        match self.payload_store.get_execution_payload(payload_id) {
-            Some(payload) => Ok(payload),
-            None => Err(EngineRpcError::UnknownPayload.into()),
-        }
+        // match self.payload_builder.get_execution_payload(payload_id) {
+        //     Some(payload) => Ok(payload),
+        //     None => Err(EngineRpcError::UnknownPayload.into()),
+        // }
```
Comment (gakonst) [inline crates/consensus/beacon/src/engine/mod.rs:293]: how will this work?

File: bin/reth/src/node/mod.rs [hunk 3]
```diff
@@ -289,8 +290,19 @@ impl Command {
...
-        let test_payload_store = TestPayloadStore::default();
+        // configure the payload builder
+        let payload_generator = BasicPayloadJobGenerator::new(
+            shareable_db.clone(),
+            transaction_pool.clone(),
+            ctx.task_executor.clone(),
+            // TODO use extradata from args
+            BasicPayloadJobGeneratorConfig::default(),
+            Arc::clone(&self.chain),
+        );
```
Comment (gakonst) [inline bin/reth/src/node/mod.rs:301]: beautiful

**Pull request merged**